src/mame/drivers/deco_mlc.cpp

```diff
@@ -267,13 +267,10 @@ READ32_MEMBER( deco_mlc_state::spriteram_r )
 
 WRITE32_MEMBER( deco_mlc_state::spriteram_w )
 {
-	if (ACCESSING_BITS_16_31)
-	{
-	}
-
+	mem_mask &= 0x0000ffff;
 	if (ACCESSING_BITS_0_15)
 	{
-		data &=0x0000ffff;
+		data &= 0x0000ffff;
 		COMBINE_DATA(&m_spriteram[offset]);
 	}
```

Comment:
What's the point of messing with the mask and data like this?  Can you make it into a 16-bit handler instead?  At any rate, if you've already restricted the mask to bits 0-15, masking the data is redundant because `COMBINE_DATA` will do that anyway.